File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/home/HomeActivity.java
package org.wikiedufoundation.wikiedudashboard.home;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import org.wikiedufoundation.wikiedudashboard.R;
import org.wikiedufoundation.wikiedudashboard.dashboard.view.MyDashboardFragment;
import org.wikiedufoundation.wikiedudashboard.helper.SharedPrefs;

import butterknife.BindView;
import butterknife.ButterKnife;

public class HomeActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {

    @BindView(R.id.drawer_layout)
    DrawerLayout drawer;

//    @BindView(R.id.tv_logout)
//    TextView tv_logout;
//
//    @BindView(R.id.tv_username)
//    TextView tv_username;

    private SharedPrefs sharedPrefs;
    private Context context;


    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = item -> {
                switch (item.getItemId()) {
                    case R.id.navigation_explore:
                        replaceFragment(new ExploreFragment());
                        return true;
                    case R.id.navigation_dashboard:
                        replaceFragment(new MyDashboardFragment());
                        return true;
                    case R.id.navigation_training:
                        replaceFragment(new TrainingFragment());
                        return true;
                }
                return false;
            };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.inflateMenu(R.menu.menu_explore);
        setSupportActionBar(toolbar);
        context = this;
        sharedPrefs = new SharedPrefs(this);
//        DrawerLayout drawer = findViewById(R.id.drawer_layout);
        NavigationView navigationView = findViewById(R.id.nav_view);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();
        navigationView.setNavigationItemSelectedListener(this);
//        tv_username.setText(sharedPrefs.getUserName());
//        tv_logout.setOnClickListener(view -> {
//            sharedPrefs.setCookies("");
//            sharedPrefs.setLogin(false);
//            sharedPrefs.setUserName("");
//            startActivity(new Intent(context, WelcomeActivity.class));
//        });

        BottomNavigationView navView = findViewById(R.id.bottom_nav_view);
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
        addFragment(new ExploreFragment());
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_explore, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_search) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        if (id == R.id.nav_settings) {
            // Handle the camera action
        }

        DrawerLayout drawer = findViewById(R.id.drawer_layout);
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }


    public void addFragment(Fragment fragment) {
        if (fragment != null) {
            FragmentManager fragmentManager = getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
//            fragmentTransaction.replace(R.id.home_layout, fragment);
//            fragmentTransaction.hide(fragmentManager.getFragments().get(fragmentManager.getBackStackEntryCount() - 1));
            fragmentTransaction.add(R.id.home_container, fragment);
            fragmentTransaction.addToBackStack(null);
//            fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_ENTER_MASK);
            fragmentTransaction.commit();
        }
    }

    public void clearStack() {
        FragmentManager manager = getSupportFragmentManager();
        if (manager.getBackStackEntryCount() > 1) {
            FragmentManager.BackStackEntry first = manager.getBackStackEntryAt(1);
            manager.popBackStack(first.getId(), FragmentManager.POP_BACK_STACK_INCLUSIVE);
        }
    }

    public void replaceFragment(Fragment fragment) {
        if (fragment != null) {
            FragmentManager fragmentManager = getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.replace(R.id.home_container, fragment);
//            fragmentTransaction.addToBackStack(null);
//            fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_ENTER_MASK);
            fragmentTransaction.commit();
        }
        if (fragment instanceof ExploreFragment){
            getSupportActionBar().show();
        }else {
            getSupportActionBar().hide();
        }
    }

}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/home/HomeActivity.kt
package org.wikiedufoundation.wikiedudashboard.ui.home

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import butterknife.ButterKnife
import com.google.android.material.bottomnavigation.BottomNavigationView
import org.koin.android.ext.android.inject
import org.wikiedufoundation.wikiedudashboard.R
import org.wikiedufoundation.wikiedudashboard.data.preferences.SharedPrefs
import org.wikiedufoundation.wikiedudashboard.ui.dashboard.view.MyDashboardFragment
import org.wikiedufoundation.wikiedudashboard.ui.profile.view.ProfileFragment

/**
 * Wikimedia homepage activity
 * ***/
class HomeActivity : AppCompatActivity() {

    private val sharedPrefs: SharedPrefs by inject()

    private var context: Context? = null
    private var myDashboardFragment: MyDashboardFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.app_bar_main)
        ButterKnife.bind(this)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        toolbar.inflateMenu(R.menu.menu_explore)
        setSupportActionBar(toolbar)
        context = this
        myDashboardFragment = MyDashboardFragment()
        val navView = findViewById<BottomNavigationView>(R.id.bottom_nav_view)
        navView.setOnNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.navigation_explore -> {
                    replaceFragment(ExploreFragment())
                    true
                }
                R.id.navigation_dashboard -> {
                    replaceFragment(myDashboardFragment)
                    true
                }
                R.id.navigation_training -> {
                    sharedPrefs.userName?.let { sharedName -> replaceFragment(ProfileFragment.newInstance(sharedName, false)) }
                    true
                }
                else -> false
            }
        }
        addFragment(MyDashboardFragment())
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_explore, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        val searchView = item?.actionView as SearchView
        searchView.queryHint = "Search"
        searchView.isIconified = false

        val txtSearch = searchView.findViewById(androidx.appcompat.R.id.search_src_text) as EditText
        txtSearch.setHintTextColor(Color.LTGRAY)
        txtSearch.setTextColor(Color.BLACK)

        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                myDashboardFragment?.updateSearchQuery(query)
                if (!searchView.isIconified) {
                    searchView.isIconified = true
                }
                item.collapseActionView()
                return false
            }

            override fun onQueryTextChange(query: String): Boolean {
                myDashboardFragment?.updateSearchQuery(query)
                return false
            }
        })
        return true
    }

    private fun addFragment(fragment: Fragment?) {
        fragment?.let {
            val fragmentManager = supportFragmentManager
            fragmentManager.beginTransaction().apply {
                add(R.id.home_container, it)
                addToBackStack(null)
                setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN)
                commit()
            }
        }
    }

    /**
     * Use [clearStack] to clear fragment back stack
     * ***/
    fun clearStack() {
        val manager = supportFragmentManager
        if (manager.backStackEntryCount > 1) {
            val first = manager.getBackStackEntryAt(1)
            manager.popBackStack(first.id, FragmentManager.POP_BACK_STACK_INCLUSIVE)
        }
    }

    private fun replaceFragment(fragment: Fragment?) {
        fragment?.let {
            val fragmentManager = supportFragmentManager
            fragmentManager.beginTransaction().apply {
                replace(R.id.home_container, fragment)
                commit()
            }
        }
        if (fragment is MyDashboardFragment) {
            supportActionBar?.show()
        } else {
            supportActionBar?.hide()
        }
    }
}
